src/animation/KeyframeTrack.js

```diff
@@ -353,6 +353,8 @@ Object.assign( KeyframeTrack.prototype, {
 	// ensure we do not get a GarbageInGarbageOut situation, make sure tracks are at least minimally viable
 	validate: function () {
 
+		if ( this.isValidated ) return true;
```

Comment:
If validation fails, it validates again next time. If we wanna avoid that, we set `undefined` to `.isValidated` as default and we change this line with?

    if ( this.isValidated !== undefined ) return this.isValidated;

Reply:
I wrote it this way because if validation fails the first time, the second time it will return false but not log any of the errors. And so for example you couldn't edit the track values in Dev Tools, re-run validation, and get useful output. Worried undefined/true/false would be more confusing. 

Redundantly validating something that isn't valid and can't be played isn't likely to hurt production performance, and it's useful in development, so I'd prefer to keep this as-is.

Reply:
Sorry for the late response.

I can understand your scenario. But feeling "Having `isValidated/Optimized` flags" while "tracks data is updatable" might be messing unless we automatically clear them when they're updated.

What I'm wondering now is whether we still need `.isValidated/Optimized` flags if we introduce the third argument `needsValidateAndOptimize` to `.clipAction()`.

(If I understand correctly) The main purpose is to skip validation and optimization for key frame animations which are already validated and optimized by the author. Just introducing `needsValidateAndOptimize` can solve it.

Reply:
> But came to think "Having isValidated/Optimized flags" while "tracks data is updatable" might be messing unless somehow we automatically clear them when they're updated.

This feels very similar to the `needsUpdate` flags we use elsewhere, which aren't automatically cleared. Would this seem better if the flags were named `needsValidate` / `needsOptimize` ?

The downside of removing the flags would be that when clips are used with clipAction twice...

```js
var action1 = mixer.clipAction( clip, mesh1 );
var action2 = mixer.clipAction( clip, mesh2 );
```

... the (sometimes expensive) validate/optimize steps will happen twice.

Reply:
Yup, that's the downside. But users can set the third arguments true.

I'm thinking if we can go first without the flags and start to consider introducing the flags when users practically request further optimization...

Reply:
We already have users [implementing their own KeyframeTrack subclass](http://assemblyltd.com/devnotes/2018/05/otg5-character-animation/) to avoid unwanted optimization calls. Today if you create two actions from a clip, optimization will only happen once. I'd rather not have this PR decrease performance for those users by optimizing twice.

Reply:
I don't think we need to fit to user defined subclass.

Reply:
Do they run multiple `.clipAction` for a `clip`? What I've seen in their article is they mentioned the constructor in `KeyframeTrack`.

Reply:
I wanna make clear. Which the issue do they wanna solve?
- validating/optimizing tracks which are already validated/optimized by the tracks author
- validating/optimizing the same track many times

Reply:
In the *Oat the Goat* blog post they want to _never_ validate or optimize clips, because they already optimized them at export time and viewers of the experience don't need to re-optimize each time they view it.

I want to provide users a way to disable all validation/optimization, like that, without hurting performance for users who don't disable it.

The tradeoff we're discussing is:

A. We hurt performance of users who create multiple AnimationActions from a single AnimationClip (e.g. reuse the same animation with multiple meshes) unless they start using this new opt-out.
B. We have `needsOptimize` / `needsValidate` flags on tracks to avoid duplicate costs, which adds a little complexity, but means if users do nothing when updating three.js, their performance will stay the same.

If we think the new `needsOptimize` / `needsValidate` are not good long-term options, but are just helpful for making this transition in a backwards-compatible way, perhaps we keep them as private undocumented flags (`_needsOptimize` / `_needsValidate`) for a few releases to warn users when redundant optimization happens? Then, eventually, remove those flags?

Reply:
Did the last part, just making the flags temporary so we can warn users about the change and later remove them.